include/psi/vm/containers/is_trivially_moveable.hpp

```diff
@@ -33,35 +33,75 @@ namespace psi::vm
+// does not, in this context, prioritize compatibility or safety - rather the
+// goal is to provide trivial relocatability support for the widest scope of
+// users/types/compilers (e.g. by using any builtin support, be it P1144 or
+// P2786) - a tracer bullet pattern/strategy - there by moving the conversation
+// further by providing more use cases and allowing the users to specialize the
+// is_trivially_moveable value for a particular type and/or compiler where the
+// OOBE is wrong/buggy.
+// @ https://github.com/psiha/vm/issues/31
+//
+// In addition, this library needs a trait like
+// 'trivially_destructible_after_move' (see vector_impl.hpp) and a trait
+// that would signal a type that contains no absolute pointers or references
+// (so that it can be trivially persisted to disk or used for IPC) - this one is
+// as of yet fully MIA and is required by vm_vector when using file backed
+// storage (until such a thingy is devised the inadequate is_trivially_moveable
+// is used).
+
 // https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2024/p1144r12.html std::is_trivially_relocatable
+// https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2024/p2786r11.html Trivial Relocatability
+// https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2023/p2959r0.html Relocation Within Containers
+// https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2024/p3055r1.html Relax wording to permit relocation optimizations in the STL
 // https://quuxplusone.github.io/blog/2019/02/20/p1144-what-types-are-relocatable
 // https://github.com/Quuxplusone/libcxx/blob/trivially-relocatable/test/libcxx/type_traits/is_trivially_relocatable.pass.cpp
 // https://brevzin.github.io/c++/2024/10/21/trivial-relocation
 // https://github.com/abseil/abseil-cpp/pull/1625
-// https://reviews.llvm.org/D114732
-// https://github.com/llvm/llvm-project/pull/88857
-// 
-// allowed to be user-specialized for custom types
+// https://reviews.llvm.org/D114732 Mark `trivial_abi` types as "trivially relocatable"
+// https://github.com/llvm/llvm-project/pull/88857 Handle trivial_abi attribute for Microsoft ABI
+// https://github.com/llvm/llvm-project/issues/69394 is_trivially_relocatable isn't correct under Windows
+// https://github.com/llvm/llvm-project/issues/86354 is_trivially_relocatable isn't correct under Apple Clang
+
+// allowed/expected to be user-specialized for custom types
 template <typename T>
 bool constexpr is_trivially_moveable
 {
 #ifdef __clang__
     __is_trivially_relocatable( T ) ||
 #endif
-#ifdef __cpp_lib_trivially_relocatable
+#if defined( __cpp_lib_trivially_relocatable /*P1144*/ ) || defined( __cpp_trivial_relocatability /*P2786*/ )
     std::is_trivially_relocatable<T> ||
 #endif
+    std::is_trivially_move_assignable_v<T> ||
+    // contrived types support below this line
     std::is_trivially_move_constructible_v<T> ||
-    std::is_trivially_copyable_v<T>
+    ( std::is_trivially_copyable_v<T> && std::is_trivially_destructible_v<T> )
```

Comment:
Nit: Any trivially copyable type _must_ be trivially destructible ([[class.prop]](https://eel.is/c++draft/class.prop#1)).
```suggestion
    std::is_trivially_copyable_v<T>
```